### PR TITLE
Safari Level

### PR DESCRIPTION
**Pull request by Farboo171**
## Description
Adds Safari Level to the Safari Zones, formula in images

## Motivation and Context
Allows locking things behind safari levels, and improvements to various safari elements based on level. Not in this PR, this is just the bones of the Level

## How Has This Been Tested?
Checked XP being gained for rock throwing, bait throwing, ball throwing, and mon catching. Checked that level does increase and the bar resets on a level.

## Screenshots (optional):
image

## Types of changes

- New feature

### CODE REVIEW

File: src/scripts/safari/Safari.ts
Comment (RegisCoaxans) [inline src/scripts/safari/Safari.ts:1]: I would recommend decreasing XP from baits and increasing XP from balls thrown since you can basically throw baits forever and for free while preventing the mon to escape too soon (and the keybind makes it AFK-compatible) whereas balls are limited per run.
image

**Review by RegisCoaxans — APPROVED** [review]

File: src/scripts/safari/Safari.ts
Comment (Farboo171) [reply src/scripts/safari/Safari.ts:1]: Swapped ball and bait XP

File: src/scripts/safari/SafariPokemonList.ts [hunk 1]
```diff
@@ -38,7 +38,7 @@ class SafariPokemonList {
         const pokemon : SafariType[] = SeededRand.shuffleArray(App.game.party.caughtPokemon.map((p) => p.name)
             .filter((p) => !PokemonHelper.hasEvableLocations(p) && Object.keys(PokemonHelper.getPokemonLocations(p)).length)
             .map((p) => {
-                return {name: p, weight: 3};
+                return {name: p, weight: 10};
```
Comment (Aegyo) [inline src/scripts/safari/SafariPokemonList.ts:41]: Was this change on purpose? It looks unrelated.

**Review by Aegyo — COMMENTED** [review]
~~I have some concerns with the safariLevel computed, but I'll need to actually measure the performance before deciding if it is worth complicating it.~~ Seems ok after testing

File: src/scripts/safari/SafariPokemonList.ts (hunk 1 above)
Comment (Farboo171) [reply src/scripts/safari/SafariPokemonList.ts:41]: This change was intentional, I should have mentioned it initially.
Filler mons in the Friend Safari were put there to keep it from breaking on strange challenge runs, but they were showing up way more frequently than felt good. This change was made to make the intended mons (Only EVable in the Friend Safari) much more common, and the filler mons be a side note.

**Review by Aegyo — APPROVED** [review]
